Add `Leading by:` feature to Leaderboard

sassypocoyo opened the pull request:
Added the `Leading by:` feature to the Leaderboard, which enables different Leaderboards based on different variables. This allowed me to create a new Distance Leaderboard, so we can now see who has placed the farthest pin from the CeSIUM Headquarters!

Also made some improvements to some functions.

**Preview:**

![image](https://user-images.githubusercontent.com/80540164/193025425-25b6caa9-03c9-4151-8a11-edd063328242.png)

RuiL1904 reviewed (commented):
I think it would make more sense if the "leading by distance" was the sum of all the distances of the user. Otherwise, this feature it's the same as "sort locations by distance descending".

sassypocoyo commented:
> I think it would make more sense if the "leading by distance" was the sum of all the distances of the user. Otherwise, this feature it's the same as "sort locations by distance descending".

It does make sense to sum the distances of all pins, it makes the "game" more interesting and takes into account the amount of pins that user has. Otherwise there would be a limit to how far a pin could be, and that author would forever be the Distance Leaderboard leader until someone traveled farther.

Anyway, implemented that and also added a Info icon to inform the user on how the distances are calculated and a hover text to the authors name saying "Go to <authors-name>'s GitHub".

nelsonmestevao commented on components/Leaderboard/style.module.css:
```diff
@@ -68,28 +68,78 @@
+  margin: 0 3px 0 3px;
+}
+
+.leading_info {
```
We should probably write our CSS module classes in PascalCase in the future :bulb:

sassypocoyo replied:
Noted, will make sure to do a PR in the future changing all CSS to PascalCase :muscle:

nelsonmestevao replied:
That would be awesome 🚀

RuiL1904 reviewed (approved):
Nice work! 👏🏼
